packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -94,6 +100,8 @@ export interface VM<N = HostNode, E = HostElement> {
     readonly owner: VM<N, E> | null;
     /** Rendering operations associated with the VM */
     readonly renderer: Renderer<N, E>;
+    /** The supported shadow flavors defined by the enum values of ShadowRenderMode */
+    shadowRenderMode: ShadowRenderMode;
```

Comment:
What about renaming this property `renderMode` and replacing the enum with?

```ts
export enum RenderMode {
    Light,
    ShadowSynthetic,
    ShadowSyntheticNative,
}
```

The `None` enum value doesn't represent well the intention to render in the Light DOM.

Reply:
I am also wondering why in this enum there isn't any Native shadow mode.

Reply:
The term "Light DOM" has always felt incorrect for me. A normal web component has both light and shadow so we basically want to say that we're removing the shadow aspect when the user chooses `renderMode = 'light'`. That's why I went with having a shadow render mode of "none".

I'm ok with changing the terminology since it's internal but would like to hear more opinions first.

Reply:
> I am also wondering why in this enum there isn't any Native shadow mode.

I used to have a native mode but I think that the only way we would select that is if we detect that the polyfill wasn't applied, and that seemed kind of leaky. I guess we already do something similar with `render.syntheticShadow` but I was actually going to replace that with comparisons against `ShadowRenderMode.Synthetic`.

Reply:
> The term "Light DOM" has always felt incorrect for me. A normal web component has both light and shadow so we basically want to say that we're removing the shadow aspect when the user chooses renderMode = 'light'. That's why I went with having a shadow render mode of "none".

I would add some nuance to _"A normal web component has both light and shadow"_. Custom elements don't need to have a shadow DOM, but also standard elements can have a shadow DOM. When setting the `LigtningELement.renderMode` you indicate to the framework that the component should be rendered in the light DOM and not in the shadow DOM. 

Now that I am thinking about it, would it make sense to split the enum into 2 enums (or replace it with a bitmask). The proposed `VM.shadowRenderMode` field currently conveys 2 information: where to render the component (light or shadow DOM) and how shadow DOM is configured (synthetic or native).